tls/s2n_internal.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "api/s2n.h"
+
+/*
+ * Internal APIs used for exploring experimental APIs.
+ *
```

Comment:
Are these necessarily "experimental"? At least for the current one, it's more "make available in rust but not in C".

Reply:
I would consider them experimental in the sense that we can change the APIs if they dont feel right. But even beyond the Rust bindings, the s2n-tls project can benefit from having a space to experiment with new APIs.